### ISSUE TITLE
Images on safari showing as broken

### ISSUE THREAD

**Issue by Chocksy** [DarthSim/imgproxy#132]
I have the following setup:

- imgproxy on a heroku dyno with the following ev variables:
```
IMGPROXY_BUGSNAG_KEY:             something...something
IMGPROXY_CONCURRENCY:             10
IMGPROXY_ENABLE_WEBP_DETECTION:   true
IMGPROXY_ENFORCE_WEBP:            true
IMGPROXY_IGNORE_SSL_VERIFICATION: true
IMGPROXY_KEY:                     something..something
IMGPROXY_MAX_CLIENTS:             20
IMGPROXY_SALT:                    something..something
IMGPROXY_WATERMARK_DATA:          something..something
```
- my server is also on heroku and reachable at www.epicpxls.com
- i have the whole thing behind cloudflare caching
- my caching is only for not logged in users but even when i login there is a random return of images that are broken: http://take.ms/vIV0A
- here is how it looks on safari in incognito not logged in: http://take.ms/KAuQR on https://www.epicpxls.com/?page=7

When i visit this same page on chrome it's all good. 
What i noticed is that when i purge all cache it gets back to normal for a few minutes but then it goes into this state again. 

Any idea what is going on? I can't figure it out. 
This is what i get when i open the image in a new tab. it just starts the download of the image and does not show it. I renamed the extension to txt so i can upload it here:
[image-7.txt](https://github.com/DarthSim/imgproxy/files/2767891/image-7.txt)
The above is for this url: https://imgproxy.epicpxls.com/oN8UmOBRCMa_xmw143TxoaWay-kP7FeyPzvYh9me9GA/rs:fill:360:270:0/g:no/aHR0cHM6Ly9pdGVt/cy5lcGljcHhscy5j/b20vdXBsb2Fkcy9w/aG90by84MjY4N2Q4/MjhjMGFiZmJjODIx/M2Y0ZmYxNWY5ZDY5/OC5wbmc.jpg

**Comment by MarcBoeing** [comment]
I found this:
https://community.cloudflare.com/t/vary-user-agent-ignored/23988

Maybe, a `vary: Accept-Encoding, Accept` may help but not sure if cf filters this.

**Comment by Chocksy** [comment]
Thank you @DarthSim & @MarcBoeing this really helped me. It seems like there is no way but to disable caching. I could make a different url but based on the browser but that's too much work for this. Seems to be working fine without the cache though for now. Thank you.

**Comment by DarthSim** [comment]
@MarcBoeing It doesn't look like CloudFlare supports `Vary`. There are many questions about it on support forum with no answers.

@Chocksy I'd recommend to turn off WebP support detection instead of turning off caching. Caching should give you more profit than WebP.

**Comment by Chocksy** [comment]
@DarthSim i see. I thought that it would actually be better to have webp but i guess you are right. Thanks a lot for the help.

**Comment by martinverdejo** [comment]
This thread has been helpful so far. I have disabled IMGPROXY_ENABLE_WEBP_DETECTION and
IMGPROXY_ENFORCE_WEBP on imgproxy config.

In addition to that my imgproxy instance is proxied on cloudflare DNS, I also needed to add a CF page rule to bypass caching on my imgproxy subdomain paths (imgproxy.sohoapp.com/* in my case). I found that CF was caching a WEBP version at that level cos I have Cf-Polish enabled. Took me a while to figure that one out.

Now more of a cloudflare issue: my imgproxy instance returns the correct format on different browsers (chrome:webp :: safari:jpeg) but when called from my cloudflare worker (assets.sohoapp.com) it always returns a webp. I think the right approach is to "configure CloudFlare to treat Accept header" as @DarthSim mentioned.
